hphp/runtime/vm/bytecode.cpp

```diff
@@ -6313,6 +6315,31 @@ void PrintTCCallerInfo() {
 static __thread Unit* s_prev_unit;
 static __thread int s_prev_line;
 
+static void recordExecutableLines(const Unit* unit, const char* filename) {
+  if (TI().m_coverage->IsRecordedExecutable(filename)) {
+    return;
+  }
+
+  auto const& sourceLocTable = getSourceLocTable(unit);
+  std::set<int> lines;
+  for (auto const& entry: sourceLocTable) {
+    auto const loc = entry.val();
+    for (int i = loc.line0; i <= loc.line1; ++i) {
+      lines.insert(i);
+    }
+  }
+  TI().m_coverage->RecordExecutable(filename, std::move(lines));
+}
+
+static void recordNotDeadCode(const Unit* unit, const char* filename) {
+  if (TI().m_coverage->IsRecordedDeadCode(filename)) {
+    return;
+  }
+
+  auto lines = unit->getLinesWithCode();
+  TI().m_coverage->RecordAsNotDeadCode(filename, std::move(lines));
+}
+
```

Comment:
These could go away, and their logic could be performed when initializing a new std::vector<int> for the file in m_hits.